keras_nlp/models/whisper/whisper_decoder.py

```diff
@@ -11,33 +11,144 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+"""Whisper decoder block."""
+
 
 from keras_nlp.backend import keras
 from keras_nlp.layers.modeling.transformer_decoder import TransformerDecoder
+from keras_nlp.models.whisper.whisper_cached_multi_head_attention import (
+    WhisperCachedMultiHeadAttention,
+)
+from keras_nlp.utils.keras_utils import clone_initializer
 
 
 @keras.saving.register_keras_serializable(package="keras_nlp")
 class WhisperDecoder(TransformerDecoder):
-    """A Whisper decoder.
+    """Whisper decoder.
 
     Inherits from `keras_nlp.layers.TransformerDecoder`, and overrides the
-    `build` method so as to remove the bias term from the key projection layer.
+    `build` method to use the
+    `keras_nlp.models.whisper.whisper_multi_head_attention.WhisperMultiHeadAttention`
+    layer instead of `keras.layers.MultiHeadAttention` and
+    `keras_nlp.models.whisper.whisper_cached_multi_head_attention.WhisperCachedMultiHeadAttention`
+    instead of `keras_nlp.layers.cached_multi_head_attention.CachedMultiHeadAttention`.
     """
 
     def build(
         self,
         decoder_sequence_shape,
         encoder_sequence_shape=None,
```

Comment:
I think we can make `encoder_sequence_shape` required and ditch the `if encoder_sequence_shape`. Whisper always has cross attention right?